src/block.ts

```diff
@@ -98,13 +104,21 @@ const produceChainBlock = async (pos: PoS, user: string, chainId: number) => {
                     reward
                 )} CTSI...`
             );
-            const gasLimit = await pos.estimateGas.produceBlock(chainId);
+
             const nonce = pos.signer.getTransactionCount("latest");
+            const gasPriceProvider = createGasPriceProvider(
+                pos.provider,
+                chainId
```

Comment:
I think there is a confusion here. The `chainId` in this context is the sidechain chainId, which starts with 0 and goes on as we can create additional chains in the future (for several reasons). It is not the Ethereum chain (mainnet, goerli, Rinkeby, etc).

It's our fault we are using the name `chainId` and causing this confusion.

Reply:
Sorry for the mix-up, I just fixed that in a new commit. I kept the `GAS_STATION_API_CHAIN_ID` config that I introduced, I think it is still relevant, but it is now checked against the provider `chainId` in the `createGasPriceProvider` function.